### PR TITLE
tree-wide: use IN_SET macro

### PR DESCRIPTION
**Pull request by yuwata**
This is a continuation of #6941 and #6933.

### CODE REVIEW

File: src/nspawn/nspawn-stub-pid1.c [hunk 1]
```diff
@@ -171,18 +171,12 @@ int stub_pid1(sd_id128_t uuid) {
                 /* Would love to use a switch() statement here, but SIGRTMIN is actually a function call, not a
                  * constant… */
 
-                if (si.si_signo == SIGRTMIN+3 ||
-                    si.si_signo == SIGRTMIN+4 ||
-                    si.si_signo == SIGRTMIN+13 ||
-                    si.si_signo == SIGRTMIN+14)
+                if (IN_SET(si.si_signo - SIGRTMIN, 3, 4, 13, 14))
```
Comment (keszybz) [inline src/nspawn/nspawn-stub-pid1.c]: I can't say I like this, I'd prefer to keep the SIGRTMIN+n syntax. We treat that as a single word everywhere.

File: src/nspawn/nspawn-stub-pid1.c [hunk 2]
```diff
@@ -171,18 +171,12 @@ int stub_pid1(sd_id128_t uuid) {
...
 
                         state = STATE_POWEROFF;
 
                 else if (si.si_signo == SIGINT ||
-                         si.si_signo == SIGRTMIN+5 ||
-                         si.si_signo == SIGRTMIN+6 ||
-                         si.si_signo == SIGRTMIN+15 ||
-                         si.si_signo == SIGRTMIN+16)
+                         IN_SET(si.si_signo - SIGRTMIN, 5, 6, 15, 16))
```
Comment (keszybz) [inline src/nspawn/nspawn-stub-pid1.c]: Here too. Let's just include all the signal numbers in a single expression.

File: src/nspawn/nspawn-stub-pid1.c (hunk 1 above)
Comment (yuwata) [reply src/nspawn/nspawn-stub-pid1.c]: Yeah. I thought it might be too aggressive. Dropped.

File: src/nspawn/nspawn-stub-pid1.c (hunk 2 above)
Comment (yuwata) [reply src/nspawn/nspawn-stub-pid1.c]: OK. Dropped.

**Comment by yuwata** [comment]
@keszybz Thank you for your review. Updated.

**Comment by keszybz** [comment]
Hmmm
```
TEST RUN: Basic systemd setup
+ /usr/bin/qemu-system-x86_64 -smp 1 -net none -m 512M -nographic -kernel /boot/vmlinuz-4.4.0-96-generic -drive format=raw,cache=unsafe,file=/var/tmp/systemd-test.RJgm4U/rootdisk.img -initrd /boot/initrd.img-4.4.0-96-generic -machine accel=kvm -enable-kvm -cpu host -append 'ro root=/dev/sda1 raid=noautodetect loglevel=2 init=/lib/systemd/systemd console=ttyS0 selinux=0   '
KVM: entry failed, hardware error 0x0
EAX=00000000 EBX=00000000 ECX=00000000 EDX=000306c1
ESI=00000000 EDI=00000000 EBP=00000000 ESP=00000000
EIP=0000fff0 EFL=00000002 [-------] CPL=0 II=0 A20=1 SMM=0 HLT=0
ES =0000 00000000 0000ffff 00009300
CS =f000 ffff0000 0000ffff 00009b00
SS =0000 00000000 0000ffff 00009300
DS =0000 00000000 0000ffff 00009300
FS =0000 00000000 0000ffff 00009300
GS =0000 00000000 0000ffff 00009300
LDT=0000 00000000 0000ffff 00008200
TR =0000 00000000 0000ffff 00008b00
GDT=     00000000 0000ffff
IDT=     00000000 0000ffff
CR0=60000010 CR2=00000000 CR3=00000000 CR4=00000000
DR0=0000000000000000 DR1=0000000000000000 DR2=0000000000000000 DR3=0000000000000000 
DR6=00000000ffff0ff0 DR7=0000000000000400
EFER=0000000000000000
Code=ff ff 66 5b 66 83 c4 08 66 5b 66 5e 66 c3 b0 20 e6 20 66 c3 <ea> 5b e0 00 f0 30 36 2f 32 33 2f 39 39 00 fc 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
autopkgtest [07:27:11]: ERROR: timed out on command "su -s /bin/bash root -c set -e; export USER=`id -nu`; . /etc/profile >/dev/null 2>&1 || true;  . ~/.profile >/dev/null 2>&1 || true; buildtree="/tmp/autopkgtest.Gk7biZ/build.N9s/debian"; mkdir -p -m 1777 -- "/tmp/autopkgtest.Gk7biZ/upstream-artifacts"; export AUTOPKGTEST_ARTIFACTS="/tmp/autopkgtest.Gk7biZ/upstream-artifacts"; export ADT_ARTIFACTS="$AUTOPKGTEST_ARTIFACTS"; mkdir -p -m 755 "/tmp/autopkgtest.Gk7biZ/autopkgtest_tmp"; export AUTOPKGTEST_TMP="/tmp/autopkgtest.Gk7biZ/autopkgtest_tmp"; export ADTTMP="$AUTOPKGTEST_TMP"; export DEBIAN_FRONTEND=noninteractive; export LANG=C.UTF-8; export DEB_BUILD_OPTIONS=parallel=1; unset LANGUAGE LC_CTYPE LC_NUMERIC LC_TIME LC_COLLATE   LC_MONETARY LC_MESSAGES LC_PAPER LC_NAME LC_ADDRESS   LC_TELEPHONE LC_MEASUREMENT LC_IDENTIFICATION LC_ALL;rm -f /tmp/autopkgtest_script_pid; set -C; echo $$ > /tmp/autopkgtest_script_pid; set +C; trap "rm -f /tmp/autopkgtest_script_pid" EXIT INT QUIT PIPE; cd "$buildtree"; export AUTOPKGTEST_NORMAL_USER=ubuntu; export ADT_NORMAL_USER=ubuntu; export 'CFLAGS=-O0'; export 'DEB_BUILD_PROFILES=noudeb'; export 'TEST_UPSTREAM=1'; export 'UPSTREAM_PULL_REQUEST=6977'; export 'GITHUB_STATUSES_URL=https://api.github.com/repos/systemd/systemd/statuses/a5fdc34dae0344ab2c5935af2dce765cd3a4860e'; chmod +x /tmp/autopkgtest.Gk7biZ/build.N9s/debian/debian/tests/upstream; touch /tmp/autopkgtest.Gk7biZ/upstream-stdout /tmp/autopkgtest.Gk7biZ/upstream-stderr; /tmp/autopkgtest.Gk7biZ/build.N9s/debian/debian/tests/upstream 2> >(tee -a /tmp/autopkgtest.Gk7biZ/upstream-stderr >&2) > >(tee -a /tmp/autopkgtest.Gk7biZ/upstream-stdout);" (kind: test)
autopkgtest [07:27:11]: test upstream: -----------------------]
autopkgtest [07:27:11]: test upstream:  - - - - - - - - - - results - - - - - - - - - -
upstream             FAIL timed out
```

I'll just restart it ;)

**Comment by yuwata** [comment]
@keszybz I found a copy-paste mistake. Updated.

**Comment by keszybz** [comment]
With SIGRTMIN the situation is nasty. Even with `-O3`, gcc is not able to coalesce all those calls. So we have code like this:
```
                i == SIGRTMIN+3 ||
   3:   e8 00 00 00 00          callq  8 <short_test+0x8>
   8:   83 c0 03                add    $0x3,%eax
                i == SIGRTMIN+4 ||
                i == SIGRTMIN+13 ||
   b:   39 d8                   cmp    %ebx,%eax
   d:   74 31                   je     40 <short_test+0x40>
                i == SIGRTMIN+4 ||
   f:   e8 00 00 00 00          callq  14 <short_test+0x14>
  14:   83 c0 04                add    $0x4,%eax
                i == SIGRTMIN+3 ||
  17:   39 d8                   cmp    %ebx,%eax
  19:   74 25                   je     40 <short_test+0x40>
                i == SIGRTMIN+13 ||
  1b:   e8 00 00 00 00          callq  20 <short_test+0x20>
  20:   83 c0 0d                add    $0xd,%eax
                i == SIGRTMIN+4 ||
  23:   39 d8                   cmp    %ebx,%eax
  25:   74 19                   je     40 <short_test+0x40>
                i == SIGRTMIN+14;
  27:   e8 00 00 00 00          callq  2c <short_test+0x2c>
  2c:   83 c0 0e                add    $0xe,%eax
                i == SIGRTMIN+13 ||
  2f:   39 d8                   cmp    %ebx,%eax
```

Maybe we should wrap SIGRTMIN in a `_const_` wrapper?

**Comment by yuwata** [comment]
@keszybz If I understand correctly, `_const_` can be used when we declare functions.

**Comment by yuwata** [comment]
The function `__libc_current_sigrtmin()` is not a constant function. See, https://github.com/bminor/glibc/blob/master/signal/allocrtsig.c.

**Comment by keszybz** [comment]
LGTM.

There's a downside though: the code is slightly larger (and presumably slower):
before:
```
-rw-rw-r--. 1 zbyszek zbyszek 5078804 Oct  4 15:57 build1/src/shared/libsystemd-shared-235.a
-rwxrwxr-x. 1 zbyszek zbyszek 4968752 Oct  4 15:57 build1/src/shared/libsystemd-shared-235.so*
```
after:
```
-rw-rw-r--. 1 zbyszek zbyszek 5091876 Oct  4 15:58 build1/src/shared/libsystemd-shared-235.a
-rwxrwxr-x. 1 zbyszek zbyszek 4976280 Oct  4 15:58 build1/src/shared/libsystemd-shared-235.so*
```
It's a few kilobytes, so not that important.

**Pull request merged**